### PR TITLE
[MMCA- 4657] - Refactor complex logic from the duty_deferment_account.scala.html in the Customs-Duty-Deferment-Frontend

### PR DESCRIPTION
**Pull request by tyagikumarvineet**
Description - Move complex logic out of the view named duty_deferment_account.scala.html

Below has been done as part of this PR

- Create new ViewModel
- update view to use the new view model
- use of new View model in Account controller
- new test class
- update existing tests
- minor refactoring

### CODE REVIEW

File: app/viewmodels/DutyDefermentAccountViewModel.scala [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+        linkMessage = "cf.accounts.older-statements.description.link",
+        location = serviceUnavailableUrl,
+        linkClass = "govuk-link govuk-link--no-visited-state",
+        preLinkMessage = Some("cf.accounts.older-statements.description")
+      )))
+  }
+
+  private def chiefDeclaration(implicit appConfig: AppConfig,
+                               messages: Messages): GuidanceRow = {
+    GuidanceRow(h2Heading = new h2().apply(
```
Comment (wg-hmrc) [inline app/viewmodels/DutyDefermentAccountViewModel.scala:213]: There is a bit of code duplication in the way elements are being created `(h2, p, link, etc.)`. Maybe consider creating utility methods to reduce this duplication.

File: test/viewmodels/DutyDefermentAccountViewModelSpec.scala [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+import views.html.requested_statements
+
+class DutyDefermentAccountViewModelSpec extends SpecBase {
+
+  "apply method" must {
+
+    "populate the model correctly" when {
+
+      "current statements are available" in new Setup {
+        implicit val appConfig: AppConfig = app.injector.instanceOf[AppConfig]
```
Comment (wg-hmrc) [inline test/viewmodels/DutyDefermentAccountViewModelSpec.scala:37]: can `appConfig` and `msgs` be in the trait object

**Review by wg-hmrc — APPROVED** [review]
LGTM

File: app/viewmodels/DutyDefermentAccountViewModel.scala (hunk 1 above)
Comment (tyagikumarvineet) [reply app/viewmodels/DutyDefermentAccountViewModel.scala]: Utility methods have been created to populate the components

File: test/viewmodels/DutyDefermentAccountViewModelSpec.scala (hunk 2 above)
Comment (tyagikumarvineet) [reply test/viewmodels/DutyDefermentAccountViewModelSpec.scala]: Moved to the trait and reusing these

**Pull request merged**